### PR TITLE
Issue #2860840 : Product variantion add to cart not translated.

### PR DESCRIPTION
**Pull request by vasike**

### CODE REVIEW

File: src/Entity/CommerceContentEntityBase.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+ * Provides the base class for Commerce content entities.
+ */
+class CommerceContentEntityBase extends ContentEntityBase implements CommerceContentEntityInterface {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getTranslatedReferencedEntities($field_name) {
+    $refereced_entities = $this->get($field_name)->referencedEntities();
+    return $this->ensureTranslations($refereced_entities);
```
Comment (GoZOo) [inline src/Entity/CommerceContentEntityBase.php]: typo error on $refereced_entities. I guess you miss a 'n' in $referenced_entities

File: src/Entity/CommerceContentEntityBase.php [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+namespace Drupal\commerce\Entity;
+
+use Drupal\Core\Entity\ContentEntityBase;
+use Drupal\Core\Language\LanguageInterface;
+
+/**
+ * Provides the base class for Commerce content entities.
+ */
+class CommerceContentEntityBase extends ContentEntityBase implements CommerceContentEntityInterface {
```
Comment (mglaman) [inline src/Entity/CommerceContentEntityBase.php]: If this is a base class, it must be abstract

File: modules/product/src/Plugin/Field/FieldWidget/ProductVariationAttributesWidget.php [hunk 3]
```diff
@@ -239,7 +239,7 @@ protected function getAttributeInfo(ProductVariationInterface $selected_variatio
       $field = $field_definitions[$field_name];
       $attributes[$field_name] = [
         'field_name' => $field_name,
-        'title' => $field->getLabel(),
+        'title' => $this->t($field->getLabel()),
```
Comment (mglaman) [inline modules/product/src/Plugin/Field/FieldWidget/ProductVariationAttributesWidget.php]: There's no need to wrap this in translatable markup, as the field is config and config is already translatable

**Review by mglaman — DISMISSED** [review]
There's a lot of changes here. And we don't have any i18n testing. There's some initial adjustments to be made.

File: src/Entity/CommerceContentEntityBase.php (hunk 2 above)
Comment (mglaman) [reply src/Entity/CommerceContentEntityBase.php]: Also, this only provides a public method for trnslated references. So really only relates to Product.

File: modules/product/src/Plugin/Field/FieldWidget/ProductVariationAttributesWidget.php (hunk 3 above)
Comment (vasike) [reply modules/product/src/Plugin/Field/FieldWidget/ProductVariationAttributesWidget.php]: i know, but there the field label won't get translated
i will update with some getTranslationFromContext somehow

File: src/Entity/CommerceContentEntityBase.php (hunk 2 above)
Comment (vasike) [reply src/Entity/CommerceContentEntityBase.php]: i think the usage could be extented on other place - for example any getStore(s) needs
and thanks

File: src/Entity/CommerceContentEntityBase.php (hunk 1 above)
Comment (vasike) [reply src/Entity/CommerceContentEntityBase.php]: indeed. merci

File: src/Entity/CommerceContentEntityBase.php (hunk 2 above)
Comment (mglaman) [reply src/Entity/CommerceContentEntityBase.php]: How would stores need this? It doesn't have an architecture which has a concept of "child references"

**Comment by mglaman** [comment]
Committed in 7af1dd81d0c4da9faebd86c941b4b0472bfa03ee

**Pull request closed without merging**